chore(release): publish 41.0.0 (#1895)
containous/traefik-helm-chart: traefik/Changelog.md

# Change Log

## 41.0.0  ![AppVersion: v3.7.5](https://img.shields.io/static/v1?label=AppVersion&message=v3.7.5&color=success&logo=) ![Kubernetes: >=1.25.0-0](https://img.shields.io/static/v1?label=Kubernetes&message=%3E%3D1.25.0-0&color=informational&logo=kubernetes) ![Helm: v3](https://img.shields.io/static/v1?label=Helm&message=v3&color=informational&logo=helm)

**Release date:** 2026-06-15

* fix(provider): :bug: emit kubernetesIngressNGINX publishService for external service
* fix(notes): :memo: use traefik.image-name so NOTES match deployed image
* fix(logs)!: align syntax with upstream (#1887)
* fix(deployment): omit spec.replicas when replicas is null
* feat(version): :sparkles: relax max-version guard to warn on minor/patch, fail only on major mismatch
* feat(providers.file)!: switch content to an object
* feat(hub): :sparkles: install out-of-box with only hub.token set
* feat(deps): update traefik docker tag to v3.7.5
* ci(renovate): restore update on appVersion
* chore(release): publish 41.0.0

**Upgrade Notes**

> There are 2 breaking changes in this release:

Author: ```suggestion
> [!IMPORTANT]
> There are 2 breaking changes in this release:
```

1. **Logs & access logs syntax**: The chart now aligns its logging keys with the upstream Traefik syntax with PR https://github.com/traefik/traefik-helm-chart/pull/1887.
   * `logs.general` is renamed to `log` (e.g. `logs.general.level` → `log.level`).
   * `logs.access` is renamed to `accessLog` (e.g. `logs.access.format` → `accessLog.format`).
   * Filter and field keys are now camelCased: `filters.statuscodes` → `filters.statusCodes`, `filters.retryattempts` → `filters.retryAttempts`, `filters.minduration` → `filters.minDuration`, and `fields.*.defaultmode` → `fields.*.defaultMode`.
   * The `accessLog.fields.general` nesting level is removed: `fields.general.defaultmode` → `fields.defaultMode`.
2. **File provider content**: `providers.file.content` is now an object (`{}`) instead of a string (`""`).
   * Migrate your inline YAML string to a structured object.

> [!TIP]

> > Image `registry` and `repository` now default to `null` and are auto-detected for Traefik Proxy & for Traefik Hub.

Reviewer: ```suggestion
> `image.registry`, `image.repository` and `image.tag` now default to `null`. The chart automatically resolves the full image URI for both Traefik Proxy (`docker.io/traefik`, tag follows the chart's `appVersion`) and Traefik Hub (`ghcr.io/traefik/traefik-hub` when `hub.token` is set, tag follows the `traefik.io/hub-max-version` annotation). If you previously hardcoded the registry, repository or tag, you can drop those overrides to let the chart keep you up to date.
```

> > Image `registry` and `repository` now default to `null` and are auto-detected for Traefik Proxy & for Traefik Hub.

Author: ```suggestion
> Image `registry` and `repository` now default to `null`.
> The chart now automatically resolves the full official image URI for both Traefik Proxy and Traefik Hub.
```

# Change Log

## 41.0.0  ![AppVersion: v3.7.5](https://img.shields.io/static/v1?label=AppVersion&message=v3.7.5&color=success&logo=) ![Kubernetes: >=1.25.0-0](https://img.shields.io/static/v1?label=Kubernetes&message=%3E%3D1.25.0-0&color=informational&logo=kubernetes) ![Helm: v3](https://img.shields.io/static/v1?label=Helm&message=v3&color=informational&logo=helm)

**Release date:** 2026-06-15

* fix(provider): :bug: emit kubernetesIngressNGINX publishService for external service
* fix(notes): :memo: use traefik.image-name so NOTES match deployed image
* fix(logs)!: align syntax with upstream (#1887)
* fix(deployment): omit spec.replicas when replicas is null
* feat(version): :sparkles: relax max-version guard to warn on minor/patch, fail only on major mismatch
* feat(providers.file)!: switch content to an object
* feat(hub): :sparkles: install out-of-box with only hub.token set
* feat(deps): update traefik docker tag to v3.7.5
* ci(renovate): restore update on appVersion
* chore(release): publish 41.0.0

> [!IMPORTANT]

> > There are 2 breaking changes in this release

Author: ```suggestion
## :warning: Breaking changes
There are 2 breaking changes in this release
```

> > There are 2 breaking changes in this release:

Author: ```suggestion
> There are 2 breaking changes in this release
```

> There are 2 breaking changes in this release:

Reviewer 2: Maybe emphasize this a bit somehow? I could see someone glossing over without catching that there are breaking changes

Author: updated. see my last commit.

> > There are 2 breaking changes in this release:

Author: ```suggestion
> [!IMPORTANT]
> There are 2 breaking changes in this release:

**Upgrade Notes**
```

1. **Logs & access logs syntax**: The chart now aligns its logging keys with the upstream Traefik syntax with PR https://github.com/traefik/traefik-helm-chart/pull/1887.
   * `logs.general` is renamed to `log` (e.g. `logs.general.level` → `log.level`).
   * `logs.access` is renamed to `accessLog` (e.g. `logs.access.format` → `accessLog.format`).
   * Filter and field keys are now camelCased: `filters.statuscodes` → `filters.statusCodes`, `filters.retryattempts` → `filters.retryAttempts`, `filters.minduration` → `filters.minDuration`, and `fields.*.defaultmode` → `fields.*.defaultMode`.

>    * The `accessLog.fields.general` nesting level is removed: `fields.general.defaultmode` → `fields.defaultMode`.

Reviewer 2: Minor nitpick: Why is the wording different?

We have:
> `logs.general` is renamed to `log` (e.g. `logs.general.level` → `log.level`)

and

> The `accessLog.fields.general` nesting level is removed: ...

Author: Because it's different. This `general` is only removed for `defaultmode` and `names`. `headers` and `queryParameters` were already under `fields`, without `general`

2. **File provider content**: `providers.file.content` is now an object (`{}`) instead of a string (`""`).

>    * Migrate your inline YAML string to a structured object.

Reviewer 2: ```suggestion
2. **File provider content**: `providers.file.content` is now an object (`{}`) instead of a string (`""`).
```
